python/shotgun_fields/image_widget.py

```diff
@@ -163,6 +187,9 @@ def setup_widget(self):
         self._view_action = QtGui.QAction("View Image", self)
         self._view_action.triggered.connect(self._show_image)
 
+        self._upload_action = QtGui.QAction("Upload Thumbnail", self)
```

Comment:
QA note: i found QActions particularily brittle when it comes to GC in some DCCs. We should add to our QA notes for this to carefully test menu operations in our various DCCs.